### PR TITLE
Predictions

### PR DESCRIPTION
**Pull request by kendevops**
_Issue #, if available:_

_Description of changes:_
I created a new `prediction.code-snippets` file and created snippets for `convert`, `identify` and `Interpret` covering all the sample cases such as `textToSpeech, transcription and so on` with `prediction configuration included`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: vscode/package.json [hunk 1]
```diff
@@ -1,4 +1,106 @@
...
+		]
+	},
+	"dependencies": {
+		"jshint": "^2.11.1",
+		"simplebuild-jshint": "^1.3.0"
+	},
+	"devDependencies": {
+		"vscode": "^1.1.18"
+	}
+=======
```
Comment (cedricgrothues) [inline vscode/package.json:103]: This seems to be a leftover from a merge conflict and causes you to have a wrong `package-lock.json` file (looks like `npm` used the monorepo `package.json`).

Comment (kendevops) [reply vscode/package.json]: Thanks for this, I have just removed it.

File: vscode/snippets/predictions.code-snippets [hunk 2]
```diff
@@ -0,0 +1,387 @@
+{
+    "Amplify Predictions Configuration": {
+		"prefix": "Predict.config",
+		"scope": "javascript,javascriptreact,typescript",
+		"body": [
+			"Amplify.configure({",
+			    "\t// To get the AWS Credentials, you need to configure",
+			    "\t// the Auth module with your Cognito Federated Identity Pool",
```
Comment (wlee221) [inline vscode/snippets/predictions.code-snippets:8]: Just my opinion, but I think we can say something like 
"You need to configure Auth (Amazon Cognito) to use the Predictions" or 
"To use Predictions, you need to configure the Auth module with your Cognito Identity Pool"
to be more explicit.

File: vscode/snippets/predictions.code-snippets [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+			    "\t\"Auth\": {",
+			        "\t\t\"identityPoolId\": ${1: \"us-east-1:xxx-xxx-xxx-xxx-xxx\"},",
```
Comment (wlee221) [inline vscode/snippets/predictions.code-snippets:10]: Remove whitespace in `${1: \"us-east-1:xxx-xxx-xxx-xxx-xxx\"}` after `1:`. Same applies for snippets below.

Comment (wlee221) [reply vscode/snippets/predictions.code-snippets:10]: You can probably use regex `[0-9]: ` to find and fix those quickly :D

File: vscode/snippets/predictions.code-snippets [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+        ]
+    },
+    "Amplify Predictions TextInImage": {
+        "prefix": "Predict.textInImage",
+        "scope": "javascript,javascriptreact,typescript",
+        "body": [
+            "Predictions.identify({",
+                "\ttext: {",
+                    "\t\tsource: {",
+                        "${1: file}",
```
Comment (wlee221) [inline vscode/snippets/predictions.code-snippets:135]: Missing tab

File: vscode/snippets/predictions.code-snippets [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+                        "\t\t\t${1: file}",
+                    "\t\t},",
+                    "\t\tformat: ${2: \"FORM\"},",
+                "\t}",
+            "})",
+            ".then(response => {",
+                "\tconst {",
+                    "\t\ttext: {",
+                        "\t\t\t// same as PLAIN +",
+                        "${3: keyValues}  // Array of { key (String), value: { text (String), selected (boolean)}, polygon, boundingBox }",
```
Comment (wlee221) [inline vscode/snippets/predictions.code-snippets:213]: Missing tab

File: vscode/snippets/predictions.code-snippets [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+    },
+    "Amplify Prediction S3 Detection": {
+        "prefix": "predict.detectFromS3",
+        "scope": "javascript,javascriptreact,typescript",
+        "body": [
+            "Predictions.identify({",
+                "\tentities: {",
+                  "\t\tsource: {",
+                    "\t\t\tkey: ${1: pathToPhoto},",
+                    "\t\t\tlevel: 'public | private | protected', //optional, default is the configured on Storage category",
```
Comment (wlee221) [inline vscode/snippets/predictions.code-snippets:272]: put space after `//`

File: vscode/snippets/predictions.code-snippets [hunk 7]
```diff
@@ -0,0 +1,387 @@
...
+                    "\t\t\tfile,",
+                  "\t\t},",
+                "\t}",
+              "})",
+              ".then((response) => console.log({ response }))",
+              ".catch(err => console.log({ err }));"
+        ]
+    },
+    "Amplify Prediction S3 Detection": {
+        "prefix": "predict.detectFromS3",
```
Comment (wlee221) [inline vscode/snippets/predictions.code-snippets:265]: `detectEntitiesFromS3` maybe

File: vscode/snippets/predictions.code-snippets [hunk 8]
```diff
@@ -0,0 +1,387 @@
...
+                    "\t\t\t ${1: width}, // ratio of overall image width",
+                    "\t\t\t ${2: height}, // ratio of overall image height",
+                    "\t\t\t ${3: left}, // left coordinate as a ratio of overall image width",
+                    "\t\t\t ${4: top} // top coordinate as a ratio of overall image height",
+                  "\t\t} = boundingBox;",
+                  "\t\tlandmarks.forEach(landmark => {",
+                      "\t\t\tconst {",
+                          " ${5: type}, // string \"eyeLeft\", \"eyeRight\", \"mouthLeft\", \"mouthRight\", \"nose\"",
+                          " ${6: x}, // ratio of overall image width",
+                          " ${7: y}, // ratio of overall image height",
```
Comment (wlee221) [inline vscode/snippets/predictions.code-snippets:303]: Missing tabs

File: vscode/snippets/predictions.code-snippets [hunk 9]
```diff
@@ -0,0 +1,387 @@
...
+        ]
+    },
+    "Amplify Prediction Detection": {
+		"prefix": "predict.detectFromBrowser",
+        "scope": "javascript,javascriptreact,typescript",
+        "body": [
+            "Predictions.identify({",
+                "\tentities: {",
+                  "\t\tsource: {",
+                    "\t\t\tfile,",
```
Comment (wlee221) [inline vscode/snippets/predictions.code-snippets:256]: Maybe we should wrap this with `${ }` as well

File: vscode/snippets/predictions.code-snippets [hunk 10]
```diff
@@ -0,0 +1,387 @@
...
+        ]
+    },
+    "Amplify Prediction Faces Detection": {
+        "prefix": "predict.detectFace",
+        "scope": "javascript,javascriptreact,typescript",
+        "body": [
+            "Predictions.identify({",
+                "\tentities: {",
+                  "\t\tsource: {",
+                    "\t\t\tfile,",
```
Comment (wlee221) [inline vscode/snippets/predictions.code-snippets:287]: Maybe we should wrap this with `${ }` as well

File: vscode/snippets/predictions.code-snippets [hunk 11]
```diff
@@ -0,0 +1,387 @@
...
+                    "\t\t\t${1: file},",
+                  "\t\t},",
+                  "\t\tcelebrityDetection: true // boolean. It will only show detected celebrities",
+                "\t}",
+            "})",
+            ".then(({ entities }) => {",
+                "\tentities.forEach(({ boundingBox, landmarks, metadata }) => {",
+                  "\t\tconst {",
+                      "\t\t\t${1: name},",
+                      "\t\t\t${1: urls}",
```
Comment (wlee221) [inline vscode/snippets/predictions.code-snippets:327]: 2:

File: vscode/snippets/predictions.code-snippets [hunk 12]
```diff
@@ -0,0 +1,387 @@
...
+                  "\t\tcollection: ${1: true}",
+                "\t}",
+            "})",
+            ".then(({ entities }) => {",
+                "\tentities.forEach(({ boundingBox, metadata: { externalImageId } }) => {",
+                  "\t\tconst {",
+                    "\t\t\t ${1: width}, // ratio of overall image width",
+                    "\t\t\t ${2: height}, // ratio of overall image height",
+                    "\t\t\t ${3: left}, // left coordinate as a ratio of overall image width",
+                    "\t\t\t ${4: top} // top coordinate as a ratio of overall image height",
```
Comment (wlee221) [inline vscode/snippets/predictions.code-snippets:354]: 2,3,4,5

File: vscode/snippets/predictions.code-snippets [hunk 13]
```diff
@@ -0,0 +1,387 @@
...
+                  "\t\t} = boundingBox;",
+                  "\t\tconsole.log({ externalImageId }); // this is the object key on S3 from the original image!",
+
+                  "\t\tStorage.get(\"\", {",
+                      "\t\t\tcustomPrefix: {",
+                      "\t\t\tpublic: externalImageId",
```
Comment (wlee221) [inline vscode/snippets/predictions.code-snippets:360]: one more tab

File: vscode/snippets/predictions.code-snippets [hunk 14]
```diff
@@ -0,0 +1,387 @@
...
+                      "\t\t\t},",
+                      "\t\t\tlevel: \"public\",",
+                  "\t\t}).then(src => console.log({ src }));",
+                "\t})",
+            "})",
+            ".catch(err => console.log(err));"
```
Comment (wlee221) [inline vscode/snippets/predictions.code-snippets:366]: Put the .catch block on line 365 for consistency

**Comment by wlee221** [comment]
LGTM, ty @kendevops!

**Pull request merged**